shared/src/prompts.rs

```diff
@@ -396,28 +396,38 @@ pub fn rename_peer(
 
 /// Presents a selection and confirmation of eligible peers for either disabling or enabling,
 /// and returns back the ID of the selected peer.
-pub fn enable_or_disable_peer(peers: &[Peer], enable: bool) -> Result<Option<Peer>, Error> {
+pub fn enable_or_disable_peer(peers: &[Peer], args: &EnableDisablePeerOpts, enable: bool) -> Result<Option<Peer>, Error> {
     let enabled_peers: Vec<_> = peers
         .iter()
         .filter(|peer| enable && peer.is_disabled || !enable && !peer.is_disabled)
         .collect();
 
-    let peer_selection: Vec<_> = enabled_peers
-        .iter()
-        .map(|peer| format!("{} ({})", &peer.name, &peer.ip))
-        .collect();
-    let (index, _) = select(
-        &format!("Peer to {}able", if enable { "en" } else { "dis" }),
-        &peer_selection,
-    )?;
-    let peer = enabled_peers[index];
+    let peer = if let Some(ref name) = args.name {
+        enabled_peers
+            .into_iter()
+            .find(|p| &p.name == name)
+            .ok_or_else(|| anyhow!("Peer '{}' does not exist", name))?
+            .clone()
+    } else {
+        let peer_selection: Vec<_> = enabled_peers
+            .iter()
+            .map(|peer| format!("{} ({})", &peer.name, &peer.ip))
+            .collect();
+        let (index, _) = select(
+            &format!("Peer to {}able", if enable { "en" } else { "dis" }),
+            &peer_selection,
+        )?;
+        enabled_peers[index].clone()
+    };
```

Comment:
Super-nit/curiosity question: previously `peer` was `&Peer`, now we've changed it to `Peer` (if I'm looking correctly), needing to `clone()`. Can we keep it a `&Peer`, perhaps if we change `.into_iter()` to `.iter()` in the first arm?

Reply:
It looks like those clones were just superfluous. Removed.